packages/commerce-sdk-react/src/hooks/ShopperCustomers/cache.ts

```diff
@@ -84,7 +84,13 @@ export const cacheUpdateMatrix: CacheUpdateMatrix<Client> = {
             remove: [{queryKey: getCustomerPaymentInstrument.queryKey(parameters)}]
         }
     },
-    deleteCustomerProductList: TODO('deleteCustomerProductList'),
+    deleteCustomerProductList(customerId, {parameters}) {
+        return {
+            // TODO: Rather than invalidate, can we selectively update?
```

Comment:
You might see these todo's left around, they are simply copied and pasted but for the most part are still valid for future cache work.

Reply:
Aside: I think we should not do this selective updates in the future. 

I've [tried](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1042) doing it in one of my PRs last time. But I reverted the change, as the logic became complex and error-prone in my experience.

Take a read of this [article](https://tkdodo.eu/blog/mastering-mutations-in-react-query#direct-updates) from one of the core maintainers of react-query. He suggested to prefer invalidation over direct updates (which is similar to "selective updates" in our terminology), because invalidation is safer and simpler approach.

Reply:
@vmarta I had the exact same thought as your suggestion, I previously said that 

> invalidation is a must have and selective updates are nice to haves

But later Alex found out an issue that made me think twice about the statement. The reason being that in SCAPI land, the models are used interchangeably across different API endpoints. For example, you get shopper's baskets from ShopperCustomer API as oppose to ShopperBasket API, but you mutate the baskets using ShopperBasket API. This creates dependencies between queries. For example, it's common that we have the following hooks on a page

1. basket mutation: addToCart
2. customer query: getCustomerBaskets

When the mutation happens, if we don't update, but only invalidate, we run into a situation that we don't have access to the invalidation fetch call, so loading state is broken for all mutations.

I'm not sure if there is another way to solve this, selective updates seems like the obvious solution...